### ISSUE TITLE
Deprecate the use of `sn.sanity_function` over the `@deferrable` builtin

### ISSUE THREAD

**Issue by jjotero** [eth-cscs/reframe#2008]
With #2007, the syntax of `sn.sanity_function` to make a function deferrable has become ambiguous with the new `sanity_function` builtin.

- Need to deprecate `sn.sanity_function`
- Probably also need to rework some of the docs replacing the terms _sanity function_ for _deferrable function_

**Comment by jjotero** [comment]
Done here https://github.com/jjotero/reframe/tree/feat/deprecate-sn-sanity-function 

Once the new syntax is merged in #2007 I'll create a new PR for this.
